vnext/Microsoft.ReactNative/Views/Image/ReactImage.cpp

```diff
@@ -213,7 +215,70 @@ void ImageFailed(const TImage &image, const TSourceFailedEventArgs &args) {
 #endif
 }
 
+// This solves the problem where images render a few frames later than adjacent UI,
+// which appear as flickers/flashes and make the app perceived as annoying and slow.
+// This method tries to render images synchronously if they are already cached
+// so that they will be painted on the screen at the same time with other UI, such as text.
+bool ReactImage::TrySetBackgroundSync(bool fireLoadEndEvent) {
```

Comment:
Thanks for keeping this out of `SetBackground` 🙂.

Room for thought: Very long functions, like `SetBackground` at over 200 lines, often create barriers to understanding, and have a whole lot of state and cycles to think about. The RNW repo enables an ESLint warning on the JS-side for functions of "high cyclomatic complexity". I wonder if we could do the same for clang-based tooling, to encourage healthy refactoring instead of "mega-functions".